add logpipeline http output docs (#14985)
kyma-project/kyma: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md

          skipCertificateValidation: true
   ```

   An output is a data destination configured by a [Fluent Bit output](https://docs.fluentbit.io/manual/pipeline/outputs) of the relevant type. The LogPipeline supports the following output types: 

   - `http`, which pushes the data to the specified http destination.
   - `custom`, which supports the configuration of any destination in the Fluent Bit configuration syntax.     

>    An example of the `custom` output is the following:

Reviewer: ```suggestion
   See the following example of the `custom` output:
```

For details, see the [LogPipeline specification file](https://github.com/kyma-project/kyma/blob/main/components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go).

| Parameter | Type | Description |
|---|---|---|
| filters | []object | List of [Fluent Bit filters](https://docs.fluentbit.io/manual/pipeline/filters) to apply to the logs processed by the pipeline. Filters are executed in sequence, as defined. They are executed before logs are buffered, and with that, are not executed on retries.|
| filters[].custom | string | Actual filter definition in the Fluent Bit syntax.|
| output | object | [Fluent Bit output](https://docs.fluentbit.io/manual/pipeline/outputs) where the logs should be pushed to. Only one output can be specified. |
| output.http | object | [Fluent Bit http output](https://docs.fluentbit.io/manual/pipeline/outputs/http). |

> | output.http.compress | string | Payload compression mechanism. |

Author: Should we add defaults? Would be helpful but then we have to be careful when they update in Fluent Bit

Reviewer 2: We should at least document where we deviate from the Fluent Bit defaults

Reviewer 3: I would add the defaults. Especially since we only expose a subset of the HTTP output fileds.

Author: added them

> | output.http.dedot | boolean | Indicates if *dedotting* (replacing dots with underscores) should be applied on the log field names. |

Reviewer 2: I think we apply that only on kubernetes.labels and kubernetes.annotations. Please double-check

Author: yes, will adjust it 👍

| output.http.format | string | Data format to be used in the HTTP request body. Default is `json`. |
| output.http.host | object | IP address or hostname of the target HTTP server. |
| output.http.host.value | string | Actual host value, can contain references to secret values. |
| output.http.host.valueFrom.secretKeyRef | object | Reference to a key in a Secret. `name` and `namespace` of the Secret, as well as the name of the `key`, must be provided. |
| output.http.password | object | Basic Auth password. |
| output.http.password.value | string | Actual password value, can contain references to secret values. |
| output.http.password.valueFrom.secretKeyRef | object | Reference to a key in a Secret. `name` and `namespace` of the Secret, as well as the name of the `key`, must be provided. |
| output.http.port | string | TCP port of the target HTTP server. Default is `443`.  |
| output.http.tls | object | TLS Configuration of the HTTP target server.  |

> | output.http.tls.disabled | boolean | Indicates if TLS is disabled or enabled. Default is `false`. |

Reviewer: Just guessing what means what. 

```suggestion
| output.http.tls.disabled | boolean | If `true`, TLS is disabled. Default is `false`, so TLS is enabled. |
```

Why on earth is this parameter called "disabled" and not "enabled" , so `true` would mean = enabled. 
Right now, we're using double negative = positive, "disabled=false" = TLS enabled, which is confusing.

Author: also thought the same  😅

Reviewer: Is it too late to change that?

Author: Reason that the name is disabled: TLS enabled should be the default, flipping the parameter to enabled, it is not possible to distinguish if the user explicitly set enabled=false or if the parameter was not set at all (and thus TLS should be active).

> | output.http.tls.skipCertificateValidation | boolean | Indicates if the validation of certificates should be skipped. Default is `false`. |

Reviewer: ```suggestion
| output.http.tls.skipCertificateValidation | boolean | If `true`, the validation of certificates is skipped. Default is `false`. |
```

| output.http.uri | string | URI for the target HTTP server. Fluent Bit Default is `/`. |
| output.http.user | object | Basic Auth username.|

> | output.http.user.value | string | Actual username value, can contain references to secret values. |

Reviewer: ```suggestion
| output.http.user.value | string | Username value. Can contain references to secret values. |
```

Is this a resource "Secret", which should be [capitalized](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#capitalization) (like line 285); or just a normal kind of secret in lower case?

> | output.http.user.valueFrom.secretKeyRef | object | Reference to a key in a Secret. `name` and `namespace` of the Secret, as well as the name of the `key`, must be provided. |

Reviewer: ```suggestion
| output.http.user.valueFrom.secretKeyRef | object | Reference to a key in a Secret. You must provide `name` and `namespace` of the Secret, as well as the name of the `key`. |
```

| output.custom | string | Any other Fluent Bit output specified in the Fluent Bit configuration syntax.|
| variables | []object | A list of mappings from Kubernetes Secret keys to environment variables. Mapped keys are mounted as environment variables, so that they are available as [Variables](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/variables) in the sections.|
| variables[].name | string | Name of the variable to map. |

> | variables[].valueFrom.secretKeyRef | object | Reference to a key in a Secret. `name` and `namespace` of the Secret, as well as the name of the `key`, must be provided. |

Reviewer: ```suggestion
| variables[].valueFrom.secretKeyRef | object | Reference to a key in a Secret. You musst provide `name` and `namespace` of the Secret, as well as the name of the `key`. |
```

| files | []object | A list of text snippets that are mounted as files to Fluent Bit, so that they are available for reference in filters and outputs. The mounted snippet is available under the `/files` folder.|
| files[].name | string | The file name under which the snippet is mounted. The resulting path will be `/files/<name>`. |
| files[].content | string | The actual text snippet to mount as file.|

### LogPipeline.status attribute

For details, see the [LogPipeline specification file](https://github.com/kyma-project/kyma/blob/main/components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go).

| Parameter | Type | Description |

> |---|---|---|

Author: just that Goland renders the tables correctly

For details, see the [LogParser specification file](https://github.com/kyma-project/kyma/blob/main/components/telemetry-operator/apis/telemetry/v1alpha1/logparser_types.go).

| Parameter | Type | Description |
|---|---|---|
| conditions | []object | An array of conditions describing the status of the parser.
| conditions[].lastTransitionTime | []object | An array of conditions describing the status of the parser.
| conditions[].reason | []object | An array of conditions describing the status of the parser.
| conditions[].type | enum | The possible transition types are:<br>- Running: The parser is ready and usable.<br>- Pending: The parser is being activated. |

> LogPipeline and LogParser examples can be found in the [samples](https://github.com/kyma-project/kyma/blob/main/components/telemetry-operator/config/samples/telemetry_v1alpha1_logparser.yaml) directory.

Reviewer: Active voice 

```suggestion
For LogPipeline and LogParser examples, see the [samples](https://github.com/kyma-project/kyma/blob/main/components/telemetry-operator/config/samples/telemetry_v1alpha1_logparser.yaml) directory.
```
